app/src/org/commcare/android/framework/SessionAwareCommCareActivity.java

```diff
@@ -1,11 +1,30 @@
 package org.commcare.android.framework;
 
+import android.os.Bundle;
+
+import org.commcare.android.util.SessionUnavailableException;
+
 /**
  * Manage redirection to login screen when session expiration occurs.
  *
  * @author Phillip Mates (pmates@dimagi.com)
  */
 public abstract class SessionAwareCommCareActivity<R> extends CommCareActivity<R> {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        try {
+            onCreateAware(savedInstanceState);
+        } catch (SessionUnavailableException e) {
+            SessionActivityRegistration.redirectToLogin(this);
+        }
```

Comment:
yeah, just calling `finish` is not a clean way out...
Basically we just need to defer until `onResume`, in which `CommCareSessionAwareActivity.onResume` will detect the loss of the session and redirect to the login screen.
Maybe this is what is bugging me: `CommCareSessionAwareActivity.onCreate`, but calling `onCreateAware`, is doing essentially what `CommCareSessionAwareActivity.onResume` is doing, aborting when the session is lost.